lib/Models/CatalogMember.js

```diff
@@ -264,11 +281,42 @@ defineProperties(CatalogMember.prototype, {
      * All keys that have historically been used to resolve this member - the current uniqueId + past shareKeys.
      */
     allShareKeys : {
-        get : function() {
+        get: function() {
             var allShareKeys = [this.uniqueId];
 
             return this.shareKeys ? allShareKeys.concat(this.shareKeys) : allShareKeys;
         }
+    },
+
+    /**
+     * A filtered view of {@link CatalogMember#info} that excludes info items that divulge details about the data's
+     * source, as determined by {@link CatalogMember#__sourceInfoItemNames}.
+     */
+    infoWithoutSources: {
+        get: function() {
+            return defaultValue(this.info, []).filter(function(infoItem) {
+                return !defined(this._infoItemsWithSourceInfoLookup[infoItem.name]);
+            }.bind(this));
+        }
+    },
+
+    /**
+     * Returns a lookup of _sourceInfoItemNames as a map of names to a true value. Memoizes after being called for the
+     * first time.
+     *
+     * {@private}
+     */
+    _infoItemsWithSourceInfoLookup: {
+        get: function() {
+            if (!defined(this._memoizedInfoItemsSourceLookup)) {
+                this._memoizedInfoItemsSourceLookup = this._sourceInfoItemNames.reduce(function(lookupSoFar, name) {
+                    lookupSoFar[name] = true;
+                    return lookupSoFar;
+                }, {});
+            }
+
+            return this._memoizedInfoItemsSourceLookup;
```

Comment:
Should you add `this._memoizedInfoItemsSourceLookup = undefined;` to the constructor?